<?xml version="1.0" encoding="UTF-8"?>

<launch>
  <arg name="folder_path"            default="/home/adam/spiirt_ws"/>
  <arg name="pcd_filename"           default="new_cloud.pcd" />
  <arg name="map_rosbag_topic"       default="grid_map" />
  <arg name="output_grid_map"        default="elevation_map.bag" />
  <arg name="map_frame"              default="map" />
  <arg name="map_layer_name"         default="elevation" />
  <arg name="prefix"                 default=""/>
  <arg name="set_verbosity_to_debug" default="false"/>

  <node pkg="spirit_utils" type="unify_gnd_map_node" name="unify_gnd_map" output="screen"/>

  <node name="grid_map_pcl_loader_node" pkg="grid_map_pcl"
      type="grid_map_pcl_loader_node" output="screen" launch-prefix="$(arg prefix)">
    <rosparam file="$(find grid_map_pcl)/config/parameters.yaml" />
    <param name="folder_path"             type="string" value="$(arg folder_path)" />
    <param name="pcd_filename"            type="string" value="$(arg pcd_filename)" />
    <param name="map_rosbag_topic"        type="string" value="$(arg map_rosbag_topic)" />
    <param name="output_grid_map"         type="string" value="$(arg output_grid_map)" />
    <param name="map_frame"               type="string" value="$(arg map_frame)" />
    <param name="map_layer_name"          type="string" value="$(arg map_layer_name)" />
    <param name="set_verbosity_to_debug"  type="bool" value="$(arg set_verbosity_to_debug)" />
  </node>


    <!-- Launch the grid map filters demo node. -->
  <node pkg="grid_map_demos" type="filters_demo" name="grid_map_filter_demo" output="screen">
    <!-- Input topic -->
  	<param name="input_topic" value="/grid_map_pcl_loader_node/grid_map_from_raw_pointcloud" />
  	<!-- Output topic -->
  	<param name="output_topic" value="filtered_map" />
  	<!-- Load grid map filter chain configuration -->
    <rosparam command="load" file="$(find grid_map_demos)/config/filters_demo_filter_chain.yaml" />
  </node>
  
  <node pkg="spirit_utils" type="terrain_map_processing_node" name="terrain_map_processing" output="screen"/>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find spirit_utils)/rviz/config.rviz"/>

</launch>
